diesel/src/mysql/connection/raw.rs

```diff
@@ -0,0 +1,97 @@
+extern crate mysqlclient_sys as ffi;
+
+use std::ffi::CStr;
+use std::os::{raw as libc};
+use std::ptr;
+use std::sync::{Once, ONCE_INIT};
+
+use result::{ConnectionResult, ConnectionError};
+use super::url::ConnectionOptions;
+
+pub struct RawConnection(*mut ffi::MYSQL);
+
+impl RawConnection {
+    pub fn new() -> Self {
+        perform_thread_unsafe_library_initialization();
+        let raw_connection = unsafe { ffi::mysql_init(ptr::null_mut()) };
+        if raw_connection.is_null() {
+            panic!("Insufficient memory to allocate connection");
+        }
+        let result = RawConnection(raw_connection);
+
+        // This is only non-zero for unrecognized options, which should never happen.
+        let charset_result = unsafe { ffi::mysql_options(
+            result.0,
+            ffi::mysql_option::MYSQL_SET_CHARSET_NAME,
+            b"utf8mb4\0".as_ptr() as *const libc::c_void,
```

Comment:
It's the equivalent of the client encoding setting in PG as best I can tell. It's the encoding used for non-query-result data. Specifically I'm setting it here because it's the encoding used for all of the parameters to `mysql_real_connect`.